src/validation/custom.validation.ts

```diff
@@ -65,26 +65,26 @@ export const checkDateIsWithinLast3Months = (errMsg: string, day: string = "", m
   return true;
 };
 
-export const checkDateValueIsValid = (invalidDateErrMsg: string, yearLengthErrMsg: string, dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+export const checkDateValueIsValid = (invalidDateErrMsg: string, dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   const day = parseInt(dayStr), month = parseInt(monthStr), year = parseInt(yearStr);
   if (isNaN(day) || isNaN(month) || isNaN(year) || !DateTime.utc(year, month, day).isValid) {
     throw new Error(invalidDateErrMsg);
-  } else if (yearStr.length !== 4) {
-    throw new Error(yearLengthErrMsg);
   }
 
   return true;
 };
 
 export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
-  if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    const areDateFieldsPresent = checkAllDateFieldsArePresent(dayStr, monthStr, yearStr);
-    if (areDateFieldsPresent) {
-      const isOptionalDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, ErrorMessages.YEAR_LENGTH, dayStr, monthStr, yearStr);
-      if (isOptionalDateValid) {
-        const isDateInThePast = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
-        if (isDateInThePast) {
-          checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+  if (yearStr.length === 0 || yearStr.length === 4) {
```

Comment:
Somebody reading this code/logic months (even years) later will probably wonder why this method doesn't check the date if the year is empty or fully populated.

Could we maybe put the `if (yearStr.length === 0 || yearStr.length === 4)` logic check in its own private function (also called by the other 'checkDate' functions below) and add a comment in that function why we only proceed if the conditions are met? I think it's because this check is done previously in the validation logic chain and we need to suppress duplicate messages if more than one condition fails?

Reply:
yes this occurred to me too after raising this PR, the month check (eg.) will only run if year is partly populated. I'm going to look at moving the if (year) check outside of these custom functions and try and have it in the body("..." validators

Reply:
updated

Reply:
Nice.